src/wallet/wallet.cpp

```diff
@@ -4315,6 +4316,76 @@ bool CWallet::BackupWallet(const std::string& strDest) const
     return GetDatabase().Backup(strDest);
 }
 
+bool CWallet::QueuedTransactionExists(const uint256& txid)
+{
+    LOCK(cs_wallet);
+    return queuedTransactionMap.count(txid) > 0;
+}
+
+bool CWallet::WriteQueuedTransaction(const uint256 &txid, const CMutableTransaction &tx)
+{
+    LOCK(cs_wallet);
```

Comment:
As far as I can tell, you actually locked the mutex already in the RPC code, right?  In this case, I guess here it should be `AssertLockHeld` instead?  (Similar for the other functions.)